### PR TITLE
error on incorrect install use

### PR DESCRIPTION
**Pull request by benjamingr**
Error on incorrect install use of lolex 2.

Fixes https://github.com/sinonjs/lolex/issues/108

### CODE REVIEW

**Comment by benjamingr** [comment]
> Could you also check the other conditions mentioned: 'config instanceof Date', arguments.length > 1 and Array.isArray?

Sure - can you elaborate a little on what that would look like? (If you just write down the rules you want me to test I'll test them).

**Comment by fatso83** [comment]
That _was_ the three rules, actually :smile:  If any of them are true, then we have a usage error. 

```javascript
if ( arguments.length > 1 || config instanceof Date || Array.isArray(config) || typeof config === 'number') {
 // throw TypeError
}
```

File: test/lolex-test.js [hunk 1]
```diff
@@ -1560,6 +1560,20 @@ describe("lolex", function () {
             assert.same(Date.clock, this.clock);
         });
 
+        it("takes an object parameter", function () {
+            this.clock = lolex.install({});
+        });
+
+        it("throws a TypeError on a number parameter", function () {
```
Comment (fatso83) [inline test/lolex-test.js]: this should be rewritten to use an assertion.

[see line 210](https://github.com/Peer5/lolex/blob/966e206c88a72c36de9561262c2f64053a2c2300/test/lolex-test.js#L210) for an example

**Comment by benjamingr** [comment]
Updated, please take a look. 

Also - any chance you could do a release today with the bugfix PR?

**Comment by fatso83** [comment]
sorry, not a chance. cell phone only in the mountains. what bugfix pr are
you talking about? this one? this is not a bug fix, but an improvement, and
it is hardly pressing. i'll look into it tomorrow.

**Comment by benjamingr** [comment]
> sorry, not a chance. cell phone only in the mountains. 

No problems - enjoy the mountains :) 

>  what bugfix pr are you talking about? this one? 

The `process.hrtime` not progressing one. 

---- 

I'd like to make sure I'm not pressuring you into anything - I know from personal experience that collaborators in open source projects are very susceptible to burn: if I'm annoying you with all the timeline questions by all means feel free to blow me off with a concrete further date I can communicate internally (like "Thursday"). 

Would hate to ruin your 🗻⛰  trip  :)

**Comment by fatso83** [comment]
Thanks!

**Comment by benjamingr** [comment]
@fatso83 do you have an estimated ETA on when 2.0.1 can happen?

**Comment by fatso83** [comment]
2.1.0 is out

**Pull request merged**